controllers/lokistack_controller.go

```diff
@@ -76,9 +80,10 @@ var (
 // LokiStackReconciler reconciles a LokiStack object
 type LokiStackReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
-	Flags  manifests.FeatureFlags
+	Log                 logr.Logger
+	Scheme              *runtime.Scheme
+	Flags               manifests.FeatureFlags
+	DependentObjectsMap map[types.NamespacedName]types.NamespacedName
```

Comment:
Why do we need a map in the first place? Adding the watch directive below should be enough to reconcile on secret changes. If the purpose is to filter secret create/update/delete events, you can certainly do this with a simple predicate rather than maintaining a custom map.

Reply:
If you look here:: https://github.com/ViaQ/loki-operator/pull/69/files#diff-5637f01ba13687b5d23f5d331aa3a063e9f3853d3dc7b1dec3911cc8d80315e8R71

The idea is that we can create a mapping between a dependent object and between some known object in lokistack. we recognize this during the first reconsile cycle. Currently, we have only one "example" of that but going forward I see more of that needed.  I will give an example - for the ruler we will want config map with rules ... this needs to redeploy just the ruler object(/s) but not the rest of the stack.

Reply:
We don't need a dependency map to solve the restart issue at hand for now. If you use the watch + predicate of controller runtime, you can restart the cluster on secret events. Regarding smart restarts (e.g. ruler only), we spiked that before and we withdrawn the idea in #47. This is not a thing we want to have right now, without understanding loki component upgrades over loki versions. We skipped this for later.

Reply:
@periklis I get it ... what I created is a very-very simple map ... actually one of the problems we will have if we drop that structure is that  the reconcile loop handles multiple lokistacks so we need to know which stack we need to reconcile on change to what secret ... I can't answer this question without that map ... 

@periklis Do you have an idea how to solve that ???